src/lex.cpp

```diff
@@ -1229,14 +1242,14 @@ void lexer::skip_block_comment() {
   QLJS_UNREACHABLE();
 
 found_comment_end:
-  this->input_ = c + 2;
+  this->input_ = html_comment ? c + 3 : c + 2;
   this->skip_whitespace();
   return;
 
   QLJS_UNREACHABLE();
 found_end_of_file:
   this->error_reporter_->report(error_unclosed_block_comment{
```

Comment:
Minor: This error type should be different for HTML comments, e.g. `error_unclosed_html_comment`.